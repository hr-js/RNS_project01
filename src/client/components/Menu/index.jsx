// @flow
import * as React from "react";
import styles from "./style.css";

type PropType = {
  current: string,
  onButtonClick: (current: string) => void
};

const menuList: Array<string> = ["入退室処理", "ユーザ登録", "利用者一覧"];

function Menu(props: PropType): React.Node {
  const { current, onButtonClick } = props;
  const { menu, list, button, selected } = styles;

  const listItems: Array<React.Node> = menuList.map(
    (item: string): React.Node => {
      const className: string =
        item === current ? `${button} ${selected}` : button;
      return (
        <li key={item}>
          <button
            className={className}
            onClick={() => {
              onButtonClick(item);
            }}
          >
            {item}
          </button>
        </li>
      );
    }
  );

  return (
    <nav className={menu}>
      <ul className={list}>{listItems}</ul>
    </nav>
  );
}
export default Menu;
