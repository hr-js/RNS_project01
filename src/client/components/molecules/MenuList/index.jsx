// @flow
import * as React from "react";
import MenuButton from "@components/atoms/MenuButton";
import styles from "./style.css";

type PropType = {
  current: string,
  onButtonClick: (current: string) => void
};

const items: Array<string> = ["入退室処理", "ユーザ登録", "利用者一覧"];

function MenuList(props: PropType): React.Node {
  const { current, onButtonClick } = props;

  const listItems: Array<React.Node> = items.map((item: string): React.Node => {
    const menuButton: React.Node =
      item === current ? (
        <MenuButton
          selected="selected"
          onButtonClick={() => {
            onButtonClick(item);
          }}
        >
          {item}
        </MenuButton>
      ) : (
        <MenuButton
          onButtonClick={() => {
            onButtonClick(item);
          }}
        >
          {item}
        </MenuButton>
      );
    return <li key={item}>{menuButton}</li>;
  });

  return <ul className={styles.menuList}>{listItems}</ul>;
}
export default MenuList;
