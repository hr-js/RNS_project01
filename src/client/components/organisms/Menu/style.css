.menu {
  position: absolute;
  width: 170px;
  height: 100%;
  top: 0;
  left: 0;
  background-color: whitesmoke;
  box-shadow: 1px 0px 4px 0px silver;
  z-index: 10;
}
