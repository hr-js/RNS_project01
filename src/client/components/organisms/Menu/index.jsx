// @flow
import * as React from "react";
import MenuList from "@components/molecules/MenuList";
import styles from "./style.css";

type PropType = {
  current: string,
  onButtonClick: (current: string) => void
};

function Menu(props: PropType): React.Node {
  const { current, onButtonClick } = props;
  return (
    <nav className={styles.menu}>
      <MenuList current={current} onButtonClick={onButtonClick} />
    </nav>
  );
}
export default Menu;
