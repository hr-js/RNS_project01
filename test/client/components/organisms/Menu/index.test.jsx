import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Menu from "@components/organisms/Menu";
import MenuList from "@components/molecules/MenuList";

describe("Menu.jsxのテスト", () => {
  const func = () => {};

  describe("スナップショット", () => {
    const tree = renderer
      .create(<Menu current="入退室処理" onButtonClick={func} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("コンポーネントのテスト", () => {
    it("MenuListを1つもつ", () => {
      const menu = shallow(<Menu current="入退室処理" onButtonClick={func} />);
      expect(menu.find(MenuList)).toHaveLength(1);
    });
  });
});
