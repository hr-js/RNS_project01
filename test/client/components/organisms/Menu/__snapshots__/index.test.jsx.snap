// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<nav
  className="menu"
>
  <ul
    className="menuList"
  >
    <li>
      <button
        className="button selected"
        onClick={[Function]}
      >
        入退室処理
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        ユーザ登録
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        利用者一覧
      </button>
    </li>
  </ul>
</nav>
`;
